editor/editor_run.cpp

```diff
@@ -101,7 +101,14 @@ Error EditorRun::run(const String &p_scene, const String p_custom_args, const Li
 			args.push_back(itos(screen_rect.position.x) + "," + itos(screen_rect.position.y));
 		} break;
 		case 1: { // centered
-			Vector2 pos = screen_rect.position + ((screen_rect.size - desired_size) / 2).floor();
+			int display_scale = 1;
+#ifdef OSX_ENABLED
+			if (OS::get_singleton()->get_screen_dpi(screen) >= 192 && OS::get_singleton()->get_screen_size(screen).x > 2000) {
```

Comment:
You could use `EDSCALE` directly AFAIK.

Reply:
And it's not only OSX which is plagued with too high res monitors ;)

Reply:
Difficult to use `EDSCALE` here, as `run/window_placement/screen` might configure the runner to a different screen than the editor (which might be different from `EDSCALE`).

From what I see in `os_windows.cpp` this seems to the macOS specific, as the other platforms are consistent with the coordinates they pass around. The complete fix might be redoing `OS_OSX` to take 2x coordinates in `OS_OSX::initialize()` and `OS_OSX::set_window_size()`, but I'm not eager to open that building site.